internal/envtest/envtest.go

```diff
@@ -3,69 +3,87 @@ package envtest
 import (
 	"context"
 	"path/filepath"
-	"runtime"
 	"strings"
 	"testing"
 
 	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
 	controlapi "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
 	k8ssandrataskapi "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 
-	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
-	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/kubectl/pkg/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 func Run(m *testing.M, setupFunc func(e *Environment)) (code int) {
-	env := NewEnvironment()
-	env.start()
+	ctx := ctrl.SetupSignalHandler()
+	env := NewEnvironment(ctx)
+	env.Start()
 	setupFunc(env)
 	exitCode := m.Run()
-	env.stop()
+	env.Stop()
 	return exitCode
 }
 
 type Environment struct {
-	intClient     client.Client
+	Client        client.Client
```

Comment:
question: Do we have a reason to export this struct value?

Reply:
Well we clearly need it in the tests... Do we have a reason not to?

Reply:
No, there's no need. We get the client with env.Client() and not with env.Client, so why would it be exported outside the pkg?

Reply:
Fine, I'm pretty sure I'll need to change this back for the next PR, but I've done as you've asked.